staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -474,14 +508,15 @@ func determineResyncPeriod(desired, check time.Duration) time.Duration {
 
 const minimumResyncPeriod = 1 * time.Second
 
-func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) {
+func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) *ResourceEventHandlerHandle {
 	s.startedLock.Lock()
 	defer s.startedLock.Unlock()
 
 	if s.stopped {
 		klog.V(2).Infof("Handler %v was not added to shared informer because it has stopped already", handler)
-		return
+		return nil
```

Comment:
if we have the possibility of returning nil, this should probably return `handle, bool` to make it clear we don't always return handles

Reply:
I added an error return here.

Reply:
let's continue this discussion with the review below